### DISCUSSION TITLE
Make Ruby on Solaris 10 fails with error No targets.  Stop. libffi-3.2.1

### DISCUSSION THREAD

**Discussion by ghost** [rbenv/ruby-build#1891] [Build failures]
gmake
..
gmake[3]: Entering directory `/export0/home/telegraf/fluentd/ruby_versions/ruby275/ruby-2.7.5/ext/fiddle/libffi-3.2.1'
gmake[3]: *** No targets.  Stop.
gmake[3]: Leaving directory `/export0/home/telegraf/fluentd/ruby_versions/ruby275/ruby-2.7.5/ext/fiddle/libffi-3.2.1'
gmake[2]: *** [libffi-3.2.1/.libs/libffi_convenience.a] Error 2
gmake[2]: Leaving directory `/export0/home/telegraf/fluentd/ruby_versions/ruby275/ruby-2.7.5/ext/fiddle'
gmake[1]: *** [ext/fiddle/all] Error 2
gmake[1]: Leaving directory `/export0/home/telegraf/fluentd/ruby_versions/ruby275/ruby-2.7.5'
gmake: *** [build-ext] Error 2
# 

/opt/csw/gnu/make
..
make[3]: Entering directory '/export0/home/telegraf/fluentd/ruby_versions/ruby275/ruby-2.7.5/ext/fiddle/libffi-3.2.1'
make[3]: *** No targets.  Stop.
make[3]: Leaving directory '/export0/home/telegraf/fluentd/ruby_versions/ruby275/ruby-2.7.5/ext/fiddle/libffi-3.2.1'
make[2]: *** [Makefile:377: libffi-3.2.1/.libs/libffi_convenience.a] Error 2
make[2]: Leaving directory '/export0/home/telegraf/fluentd/ruby_versions/ruby275/ruby-2.7.5/ext/fiddle'
make[1]: *** [exts.mk:228: ext/fiddle/all] Error 2
make[1]: Leaving directory '/export0/home/telegraf/fluentd/ruby_versions/ruby275/ruby-2.7.5'
make: *** [Makefile:983: build-ext] Error 2
# 
=== debug:

GNU Make 4.3
Built for i386-pc-solaris2.10
Copyright (C) 1988-2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Reading makefiles...
Updating makefiles....
Updating goal targets....
 File 'all' does not exist.
   File 'install' does not exist.
     File 'install-so' does not exist.
       File '../../.ext/i386-solaris2.10/fiddle.so' does not exist.
         File 'libffi-3.2.1/.libs/libffi_convenience.a' does not exist.
        Must remake target 'libffi-3.2.1/.libs/libffi_convenience.a'.
make[2]: Entering directory '/export0/home/telegraf/fluentd/ruby_versions/ruby275/ruby-2.7.5/ext/fiddle'
GNU Make 4.3
Built for i386-pc-solaris2.10
Copyright (C) 1988-2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Reading makefiles...
Updating makefiles....
make[3]: Entering directory '/export0/home/telegraf/fluentd/ruby_versions/ruby275/ruby-2.7.5/ext/fiddle/libffi-3.2.1'
make[3]: *** No targets.  Stop.
make[3]: Leaving directory '/export0/home/telegraf/fluentd/ruby_versions/ruby275/ruby-2.7.5/ext/fiddle/libffi-3.2.1'
make[2]: *** [Makefile:377: libffi-3.2.1/.libs/libffi_convenience.a] Error 2
make[2]: Leaving directory '/export0/home/telegraf/fluentd/ruby_versions/ruby275/ruby-2.7.5/ext/fiddle'
make[1]: *** [exts.mk:228: ext/fiddle/all] Error 2
make[1]: Leaving directory '/export0/home/telegraf/fluentd/ruby_versions/ruby275/ruby-2.7.5'
make: *** [Makefile:983: build-ext] Error 2
#

**Comment by hsbt** [comment]
It's not ruby-build issue.
